fenicsadapter/fenicsadapter.py

```diff
@@ -1,759 +1,458 @@
-
-        :param x: coordinate where expression has to be evaluated
-        :param value: buffer where result has to be returned to
+    def create_point_sources(self, fixed_boundary, data=None):
+        """
+        Create point sources with reference to fixed boundary in a FSI simulation.
+
+        Parameters
+        ----------
+        fixed_boundary : FEniCS domain
+            FEniCS domain consisting of a fixed boundary condition. For example in FSI cases usually the solid body
+            is fixed at one end.
+        data : PointSource
+            FEniCS PointSource data.
+
+        Returns
+        -------
+        x_forces : list
+            List containing X component of forces with reference to respective point sources on the coupling interface.
+        y_forces : list
+            List containing Y component of forces with reference to respective point sources on the coupling interface.
         """
-        return_value = self.interpolate(x)
-        for i in range(self._vals.ndim):
-            value[i] = return_value[i]
+        if not data:
+            data = np.zeros((self._n_vertices, self._fenics_dimensions))
 
-    def is_scalar_valued(self):
-        """ Determines if function being interpolated is scalar-valued based on dimension of provided vector self._vals.
+        self._Dirichlet_Boundary = fixed_boundary
+        return self.update_point_sources(data)
 
-        :return: whether function is scalar valued
+    def update_point_sources(self, data):
         """
-        if self._vals.ndim == 1:
-            return True
-        elif self._vals.ndim > 1:
-            return False
-        else:
-            raise Exception("Dimension of the function is 0 or negative!")
+        Update values of point sources using data. This function only works for 2D-pseudo3D coupling.
+
+        Parameters
+        ----------
+        data : PointSource
+            FEniCS PointSource data.
+
+        Returns
+        -------
+        x_forces : list
+            List containing X component of forces with reference to respective point sources on the coupling interface.
+        y_forces : list
+            List containing Y component of forces with reference to respective point sources on the coupling interface.
+        """
+        return get_forces_as_point_sources(self._Dirichlet_Boundary, self._function_space, self._coupling_mesh_vertices,
+                                           data)
 
-    def is_vector_valued(self):
-        """ Determines if function being interpolated is vector-valued based on dimension of provided vector self._vals.
+    def read_data(self):
+        """
+        Read data from preCICE. Data is generated in an appropriate form depending on the dimensions of the
+        simulation (2D-3D Coupling, 2D-2D coupling or Scalar/Vector write function).
+        Note: For quasi 2D fenics in a 3D coupled simulation the y component of the vectors is deleted.
```

Comment:
What does "appropriate format" mean? Usually this function is used together with `update_point_sources` or `update_coupling_boundary_conditions`, correct? Mentioning that "appropriate" data refers to providing data that is fit for use with these functions might help.